app/components/map-interface/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { makeStyles, useTheme } from "@rneui/themed"
+import MapView, { Region } from "react-native-maps"
+import MapStyles from "./map-styles.json"
+import React, { useRef } from "react"
+import { BusinessMapMarkersQuery } from "@app/graphql/generated"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import MapMarker from "../map-marker"
+
+type Props = {
+  data?: BusinessMapMarkersQuery
+  userLocation?: Region
+  handleMapPress: () => void
+  handleMarkerPress: (_: MarkerData) => void
+  focusedMarker: MarkerData | null
+  handleCalloutPress: (_: MarkerData) => void
+}
+
+export type MarkerData = {
+  readonly __typename: "MapMarker"
+  readonly username?: string | null | undefined
+  readonly mapInfo: {
+    readonly __typename: "MapInfo"
+    readonly title: string
+    readonly coordinates: {
+      readonly __typename: "Coordinates"
+      readonly longitude: number
+      readonly latitude: number
+    }
+  }
+}
+
+export default function MapInterface({
+  data,
+  userLocation,
+  handleMapPress,
+  handleMarkerPress,
+  focusedMarker,
+  handleCalloutPress,
+}: Props) {
+  const {
+    theme: { colors, mode: themeMode },
+  } = useTheme()
+  const styles = useStyles()
+  const { LL } = useI18nContext()
+  const text = React.useMemo(() => LL.MapScreen.payBusiness(), [LL.MapScreen])
+
+  const mapViewRef = useRef<MapView>(null)
+
+  return (
+    <MapView
+      ref={mapViewRef}
+      style={styles.map}
+      showsUserLocation={true}
+      initialRegion={userLocation}
+      customMapStyle={themeMode === "dark" ? MapStyles.dark : MapStyles.light}
+      onPress={handleMapPress}
+    >
+      {(data?.businessMapMarkers ?? []).reduce(
+        (arr: React.ReactElement[], item: MarkerData | null) => {
+          if (item?.username) {
```

Comment:
I think right now by definition we have a username for the every point on the map - although it might not be reflected in the type?

Reply:
The way everything was written before made it very clear that there will some items that were null. I followed suit with that in mind, so there's several instances where I watch out for it. The reduce function instead of map which you mentioned below is an example.

If you're sure they're all going to be populated now then I can certainly remove these kind of checks

Reply:
making a type nullable to non nullable is not a breaking change (the other way around is). let me look into that before we merge the PR